samples/TrayIcon/AboutWindow.axaml

```diff
@@ -0,0 +1,18 @@
+<Window xmlns="https://github.com/avaloniaui"
```

Comment:
I am not sure about adding yet another project here.
Ideally, TrayIcon test should be added to the IntegrationTestApp project + new e2e tests written per each platform (can be done separately).

Reply:
Hmm, okay. Coming at this from the other direction, when I first started writing my application, I found all existing documentation of `TrayIcon` to be incomplete and there were no samples. This would be _the first_ sample showing how to use `TrayIcon` as far as I am aware. It serves to illustrate the crash in this PR, but after the PR is merged, it serves as a `TrayIcon` example for new developers.

Reply:
Is there a coded UI test infrastructure in place that would allow automated testing of this? If so then it's also a good point to add a test of this crash to that as well. :-)

Reply:
Hmm, I duplicated the `TrayIcon` stuff into `IntegrationTestApp` but no menu appears when I click on the tray icon. I tried a bunch of different ways to make `IntegrationTestApp` be more like `TrayIcon` but I wasn't able to figure out what the difference is.

Reply:
I see project `Avalonia.IntegrationTests.Appium`. Does Appium support interacting with tray icons? I tried some searches related to this and couldn't find anything conclusive. I've never worked with Appium before.

Reply:
I am not familiar with Appium coded UI tests, and to me it feels like a significant increase in scope to replace a straightforward sample app with automated tests. Perhaps that could be captured in a backlog for someone with the expertise and handled independently from this PR?

Reply:
> I am not sure about adding yet another project here.

Regarding this point, I personally feel like it makes sense, at least from my outsiders' perspective. The folder is called `samples`. I am a new developer. I want to know how to do a tray icon. I stumble across `samples`. If one of them says `TrayIcon`, then I know exactly where to go. If not, then how am I supposed to just magically know that `IntegrationTestApp` is what I should be looking at? And how is an integration test test app a good sample of how the functionality works??

Reply:
> This would be the first sample 

ControlCatalog has one. But yeah, we should have proper docs and Avalonia.Samples as well.

`IntegrationTestApp` is a normal Avalonia app, so tray icon should work there. `Avalonia.IntegrationTests.Appium` is a e2e testing project which runs `IntegrationTestApp` app, and automatically manipulates with the app to check all the test cases. 

> Does Appium support interacting with tray icons

That's a good question. It likely does, but might be tricky to get correct automation right.

Reply:
> Perhaps that could be captured in a backlog for someone with the expertise and handled independently from this PR?

Yeah, as I mentioned in the first message, e2e can be done separately. But let's avoid another project in this PR too.

Reply:
Hmm, okay. That project is serving a purpose presently, though -- it's fulfilling the "first demonstrate the problem in one commit" part before the next commit fixes the problem. You want I should edit that commit out anyway, so that this PR is just a bug fix out of the blue with no proof?

Reply:
If you would prefer to have any sample, please add it to the `IntegrationTestApp` project. Without any tests, just a tray icon. Somebody else, probably me, will add tests on top of that later.

Reply:
Hmm, well, I tried to merge it in there, but when I run the `IntegrationTestApp` project, the menu doesn't show up when clicking on the tray icon. Not sure how it's different from the `TrayIcon` sample, but there must be something...
```
    <TrayIcon.Icons>
        <TrayIcons>
            <TrayIcon Icon="/Assets/icon.ico">
                <TrayIcon.Menu>
                    <NativeMenu>
                        <NativeMenuItem Header="Show _Test Window" Command="{Binding ShowWindowCommand}" />
                    </NativeMenu>
                </TrayIcon.Menu>
            </TrayIcon>
        </TrayIcons>
    </TrayIcon.Icons>
```

Reply:
Works here. Right click. But I removed command as it doesn't have view model locally.

![image](https://github.com/AvaloniaUI/Avalonia/assets/3163374/6b6dc308-9712-4622-ad62-46a6788d0ab9)

Reply:
That's very strange. It continues to not work for me. I tried removing the command binding just in case that was why but it didn't fix it. Since I can't get the menu to show up, I can't confirm the crash with `IntegrationTestApp`. It's a technicality but I would feel better if you could confirm the crash without c71d999 and the fix with it.

Reply:
Or, rather, with and without 04e3225ff. :-)

Reply:
Okay I switched from GNOME Desktop to KDE Plasma and now the menu shows up, and I can confirm the fix. :-) Definitely crashes without 04e3225, definitely fixed with, at least on my end.